modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/LeftSidebar/LeftSidebarTreeView.tsx

```diff
@@ -225,6 +225,22 @@ export default function LeftSidebarTreeView({
 				<TreeView.Item>
 					<TreeView.ItemStack>
 						<div className="lfr-objects__model-builder-left-sidebar-current-object-folder-container">
+							<div className="lfr-objects__model-builder-left-sidebar-current-object-collapse-icon">
+								<ClayButtonWithIcon
+									aria-label={
+										expanded
+											? Liferay.Language.get('collapse')
+											: Liferay.Language.get('expand')
+									}
+									displayType="unstyled"
+									onClick={() => setExpanded(!expanded)}
+									size="xs"
+									symbol={
+										expanded ? 'angle-right' : 'angle-down'
+									}
+								/>
+							</div>
+
```

Comment:
Hey @Artartur, We can adopt another simpler solution, the TreeView component uses a `nestedKey` prop to see if the `items` have any type of "children" to see if it will be expanded or not, but this prop must have the same name that these "children" have in JSON that we are passing on to him. If you go to `LeftSidebarTreeView.tsx` on line `179` we are passing the `items` to the TreeView, analyzing this `newLeftSidebarOtherObjectFolderItems` that we are passing I could see that the name of these "children" is `leftSidebarObjectDefinitionItems`, so the only thing you need to do is put exactly the same name in the `nestedKey` and the component will be handled automatically. :smile: 

```tsx
nestedKey="leftSidebarObjectDefinitionItems"
```

Reply:
I got it, done!. Thx @danvb112!